flink-examples/src/main/java/io/pravega/examples/flink/alert/HighCountAlerter.java

```diff
@@ -0,0 +1,206 @@
+
+        // add the Pravega reader as the data source
+        DataStream<String> inputStream = env.addSource(reader);
+
+        // create an output sink to stdout for verification
+        //inputStream.print();
+
+        // transform logs
+        DataStream<AccessLog> dataStream = inputStream.map(new ParseLogData());
+
+        // create an output sink to stdout for verification
+        //dataStream.print();
+
+        // get responses and their counts
+        DataStream<ResponseCount> countStream =
+            dataStream.flatMap(new FlatMapFunction<AccessLog, ResponseCount>() {
+                @Override
+                public void flatMap(AccessLog value, Collector<ResponseCount> out) throws Exception {
+                    out.collect(new ResponseCount(value.getStatus(), 1));
+                }
+            }).filter((FilterFunction<ResponseCount>) count -> {
+                  return !count.response.isEmpty();
+              }).keyBy("response")
+              .timeWindow(Time.seconds(Constants.ALERT_WINDOW), Time.seconds(Constants.ALERT_INTERVAL))
+              .sum("count");
+
+        // create an output sink to stdout for verification
+        countStream.print();
+
+        // create alert pattern
+        Pattern<ResponseCount,?> pattern500 = Pattern.<ResponseCount>begin("500pattern")
+            .where(new SimpleCondition<ResponseCount>() {
+                @Override
+                public boolean filter(ResponseCount value) throws Exception {
+                    return value.count >= Constants.ALERT_THRESHOLD &&
+                           value.response.equals("500");
+                }
+            });
+
+        PatternStream<ResponseCount> patternStream = CEP.pattern(countStream, pattern500);
+
+        DataStream<Alert> alertStream = patternStream.select(
+            new PatternSelectFunction<ResponseCount, Alert>() {
+                @Override
+                public Alert select(Map<String, List<ResponseCount>> pattern) throws Exception {
+                    ResponseCount count = pattern.get("500pattern").get(0);
+                    return new Alert(count.response, count.count, "High 500 responses");
+                }
+            });
+
+        // create an output sink to stdout for verification
+        alertStream.print();
+
+
+        // execute within the Flink environment
+        env.execute("HighCountAlerter");
+
+        LOG.info("Ending HighCountAlerter...");
+    }
+
+    //Parse the incoming streams & convert into Java PoJos
+    private static class ParseLogData implements MapFunction<String, AccessLog>{
+        public AccessLog map(String record) throws Exception {
+            // TODO: handle exceptions
+            Gson gson = new Gson();
+            AccessLog accessLog = new AccessLog();
+            JsonParser parser = new JsonParser();
+            JsonObject obj = parser.parse(record).getAsJsonObject();
```

Comment:
I think Jersey Jackson module (ObjectMapper) can help convert json string to bean if we want to avoid the boiler plate code?

Reply:
Done. Use jackson now.